plugin/cache.go

```diff
@@ -134,7 +149,11 @@ func ClearCacheFor(path string) ([]string, error) {
 	return cache.Delete(rx), nil
 }
 
-func cachedOpHelper(op cachedOp, entry Entry, id string, generateValue func() (interface{}, error)) (interface{}, error) {
+func panicNilCacheConfig() {
```

Comment:
I wonder if we can just embed `CacheConfig` in the `Entry` interface (i.e. have its methods be part of the `Entry` interface). We can force plugin authors to use EntryBase everywhere (and use `plugin.NewEntry` to create these objects). Embedding `CacheConfig` in `Entry` would avoid the nil cache config case, and it makes it easy to reason about stuff (all entries are cacheable and have cache IDs, but you can choose to turn off caching if you don't need the behavior)

Reply:
Actually, we could just have `GetTTLOf`, and `ID` added to Entry while the setters could be restricted to EntryBase.

I don't see a good reason why plugin authors shouldn't use EntryBase all the time.

Reply:
That seems ok.

Reply:
I'll do that in this PR as a separate commit